var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -189,6 +190,8 @@ class Dd4hep(CMakePackage):
     )
     conflicts("~ddrec+dddetectors", msg="Need to enable +ddrec to build +dddetectors.")
 
+    conflicts("geant4 cxxstd=11", when="@1.27.1: +ddg4")
```

Comment:
I'd remove the `when @1.27.1:`, this seems like it could nudge the concretizer to build dd4hep 1.27.0 when that's not really what you want.

Reply:
That is a good idea. I have also added the gist of the discussion we had in here as a comment.